### ISSUE TITLE
Upgrade to Gradle 8

### ISSUE THREAD

**Issue by mh182** [docToolchain/docToolchain#1103]
docToolchain uses Gradle 6.9.2. We have to upgrade Gradle to a version which supports Java 17, a precondition for #1071.

With this issue, we upgrade to the newest release of [Gradle 8](https://gradle.org/whats-new/gradle-8/).

[Deprecation Warnings for an upgrade to Gradle 7](https://scans.gradle.com/s/hran3rnxoehr4/deprecations):
```
➜ ./gradlew help --warning-mode=all

> Configure project :
The testCompile configuration has been deprecated for dependency declaration. This will fail with an error in Gradle 7.0. Please use the testImplementation configuration instead. Consult the upgrading guide for further information: https://docs.gradle.org/6.9.2/userguide/upgrading_version_5.html#dependencies_should_no_longer_be_declared_using_the_compile_and_runtime_configurations
        at build_7f3j70txdw36gw7pdzvrurdch$_run_closure1.doCall(/home/max/dev/dtc/docToolchain/build.gradle:42)
        (Run with --stacktrace to get the full stack trace of this deprecation warning.)
The testRuntime configuration has been deprecated for dependency declaration. This will fail with an error in Gradle 7.0. Please use the testRuntimeOnly configuration instead. Consult the upgrading guide for further information: https://docs.gradle.org/6.9.2/userguide/upgrading_version_5.html#dependencies_should_no_longer_be_declared_using_the_compile_and_runtime_configurations
        at build_7f3j70txdw36gw7pdzvrurdch$_run_closure1.doCall(/home/max/dev/dtc/docToolchain/build.gradle:45)
        (Run with --stacktrace to get the full stack trace of this deprecation warning.)
The AbstractArchiveTask.destinationDir property has been deprecated. This is scheduled to be removed in Gradle 7.0. Please use the destinationDirectory property instead. See https://docs.gradle.org/6.9.2/dsl/org.gradle.api.tasks.bundling.AbstractArchiveTask.html#org.gradle.api.tasks.bundling.AbstractArchiveTask:destinationDir for more details.
        at build_7f3j70txdw36gw7pdzvrurdch$_run_closure5.doCall(/home/max/dev/dtc/docToolchain/build.gradle:176)
        (Run with --stacktrace to get the full stack trace of this deprecation warning.)
```

#662 was started but never finished and would be obsolete by this task.

**Comment by PacoVK** [comment]
This is blocked by https://github.com/asciidoctor/asciidoctor-gradle-plugin/issues/664

**Comment by PacoVK** [comment]
prepared in #1104  going to 7.0.2

**Comment by mh182** [comment]
> This is blocked by [asciidoctor/asciidoctor-gradle-plugin#664](https://github.com/asciidoctor/asciidoctor-gradle-plugin/issues/664)

I created #1105 so we could test out, if this blocker still exists in v4.*.

**Comment by mh182** [comment]
@PacoVK wrote in [comment of #1105](https://github.com/docToolchain/docToolchain/issues/1105#issuecomment-1506747430)
> Sure I will do that.
> 
> Concerning Gradle version unfortunately there is an incompatibility from 7.6 onwards with the Asciidoctor Gradle plugin see [issue](https://github.com/asciidoctor/asciidoctor-gradle-plugin/issues/655) I would not suggest to try some of the mentioned workarounds but rather wait for the Gradle 8 support and just skip 7.6 then

So this is on hold until upstream provides a fix.
